pkg/discovery-manager/discovery_controller.go

```diff
@@ -47,13 +56,276 @@ type DiscoveryReconciler struct {
+				peeringCandidate = createPeeringCandidateCustomResources(flavourCR, &discovery, first, i)
+				klog.Infof("Peering Candidate %s created", peeringCandidate.Name)
+			} else {
+				// the others are created as not reserved
+				peeringCandidateNotReserved := createPeeringCandidateCustomResources(flavourCR, &discovery, first, i)
+				klog.Infof("Peering Candidate %s created but not reserved", peeringCandidateNotReserved.Name)
+				err = r.Create(context.Background(), peeringCandidateNotReserved)
+				if err != nil {
+					klog.Errorf("Error when creating Peering Candidate not reserved: %s", err)
+				}
+			}
+			i++
+			first = false
+		}
+
+		err = r.Create(context.Background(), peeringCandidate)
+		if err != nil {
+			klog.Errorf("Error when creating Peering Candidate: %s", err)
+			setDiscoveryPhase(&discovery, nodecorev1alpha1.PhaseFailed)
+		} else {
+			// Update the Discovery with the PeeringCandidate
+			discovery.Spec.PeeringCandidate.Name = peeringCandidate.Name
+			discovery.Spec.PeeringCandidate.Namespace = DEFAULT_NAMESPACE
+			if err := r.Update(ctx, &discovery); err != nil {
+				klog.Errorf("Error when updating Discovery %s: %s", discovery.Name, err)
+				return ctrl.Result{}, err
+			}
+			klog.Infof("Discovery %s updated", discovery.Name)
+			setDiscoveryPhase(&discovery, nodecorev1alpha1.PhaseSolved)
+		}
+		if err := r.updateSolverStatus(ctx, &discovery); err != nil {
+			klog.Errorf("Error when updating Solver %s status: %s", req.NamespacedName, err)
+			return ctrl.Result{}, err
+		} else {
+			// Create a Reservation CR
+			reservationCR := createReservationCustomResource(peeringCandidate)
+			err := r.Create(context.Background(), reservationCR)
+			if err != nil {
+				klog.Errorf("Error when creating Reservation: %s", err)
+				setDiscoveryPhase(&discovery, nodecorev1alpha1.PhaseFailed)
+			} else {
+				klog.Infof("Reservation %s created", reservationCR.Name)
+			}
+			return ctrl.Result{}, err
+		}
+	case nodecorev1alpha1.PhaseSolved:
+
+		klog.Infof("Discovery %s solved", discovery.Name)
+		setDiscoveryPhase(&discovery, nodecorev1alpha1.PhaseSolved)
+		if err := r.updateSolverStatus(ctx, &discovery); err != nil {
+			klog.Errorf("Error when updating Solver %s status: %s", req.NamespacedName, err)
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	case nodecorev1alpha1.PhaseFailed:
+		klog.Infof("Discovery %s failed, no Peering Candidate found", discovery.Name)
+		setDiscoveryPhase(&discovery, nodecorev1alpha1.PhaseFailed)
+		if err := r.updateSolverStatus(ctx, &discovery); err != nil {
+			klog.Errorf("Error when updating Solver %s status: %s", req.NamespacedName, err)
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
 
 	return ctrl.Result{}, nil
 }
 
+// setDiscoveryPhase sets the phase of the discovery
+func setDiscoveryPhase(d *advertisementv1alpha1.Discovery, phase nodecorev1alpha1.Phase) {
+	d.Status.Phase.Phase = phase
```

Comment:
The update time? take inspiration from my work on REAR Manager

Reply:
Fixed.